src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,45 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  padding: $padding-size;
+  gap: $gap-size;
+  width: $width;
+  box-sizing: border-box;
+
+  &__day {
+    background-color: #efe9e9;
+    height: $day-box-size;
+    width: $day-box-size;
+    border: $border-size solid #000;
+    box-sizing: border-box;
+    font-family: Arial, sans-serif;
+    font-size: 30px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transition: transform 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i}):before {
+        content: '#{$i}';
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      transform: translateY(-20px);
+    }
+  }
+
+  @each $day, $marg in $start-day {
+    &--start-day-#{$day} :first-child {
+      margin-left: $marg;
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} :nth-child(n + #{$i + 1}) {
```

Comment:
There is an extra space before ':nth-child'. According to the code style rules, there should be no space between the element and the pseudo-class.